crates/wit-parser/src/metadata.rs

```diff
@@ -521,6 +521,7 @@ struct InterfaceMetadata {
         serde(default, skip_serializing_if = "StringMap::is_empty")
     )]
     types: StringMap<TypeMetadata>,
+    nested: StringMap<NestMetadata>,
```

Comment:
Could this include the attribute to skip serializing if this is empty? I think that'll assist with backwards compatibility by not breaking historical decoders that don't have this field when it's not used.